sdl_ios/SmartDeviceLink/SDLDataStreamHandlingDelegate.m

```diff
@@ -0,0 +1,53 @@
+//
+//  SDLDataStreamHandlingDelegate.m
+//  SmartDeviceLink
+//
+
+#import <objc/runtime.h>
+#import "SDLProtocolMessage.h"
+#import "SDLDataStreamHandlingDelegate.h"
+#import "SDLDebugTool.h"
+
+@implementation SDLDataStreamHandlingDelegate
+
+- (void)stream:(NSStream *)stream handleEvent:(NSStreamEvent)eventCode
+{
+
+    switch(eventCode)
+    {
+        case NSStreamEventHasBytesAvailable:
+        {
+            // Grab some bytes from the stream
+            const int bufferSize = 1024;
+            uint8_t buf[bufferSize];
+            NSUInteger len = [(NSInputStream *)stream read:buf maxLength:bufferSize];
+            if(len > 0)
+            {
+                NSData* data = [NSData dataWithBytes:buf length:len];
+                [self.protocol sendRawData:data withServiceType:self.serviceType];
```

Comment:
I don't understand why we want developer to pass in an input stream if all we're going to do is read and send the raw data. They have the full file, why don't they just pass it to us, a file handle for the file, or the raw data itself?